ansible/roles/operator-pipeline/templates/openshift/tasks/set-github-status.yml

```diff
@@ -0,0 +1,83 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: set-github-status
+spec:
+  params:
+    - name: git_repo_url
+    - name: commit_sha
+      description: SHA of the commit to set the status for.
+    - name: description
+      description: Description attached to the github status.
+    - name: state
+      default: ""
+      description: The github status to set for the given commit.
+    - name: context
+      description: Context attached
+    - name: preflight_results_exists
+      default: "false"
+    - name: result_file
+      default: ""
+      description: Path to the file that contains test results.
+    - name: result_url
+      default: ""
+      description: URL to the test result on Pyxis.
+  workspaces:
+    - name: github-bot-token
+    - name: source
+  steps:
+    - name: set-github-status
+      image: quay.io/redhat-isv/operator-pipelines-images:latest
+      workingDir: $(workspaces.source.path)
+      script: |
+        #! /usr/bin/env bash
+        # Don't use set -x to avoid exposing github token
+        set -e
+
+        PREFLIGHT_RESULTS_EXISTS="$(params.preflight_results_exists)"
+        if [ $PREFLIGHT_RESULTS_EXISTS == "true" ]; then
+          echo "Preflight results already exists - skipping setting github status"
+          exit 0
+        fi
+
+        DESCRIPTION="$(params.description)"
+
+        if [ -f "$(params.result_file)" ]; then
```

Comment:
This part of the task that you wrote does the job, that will be done in ISV-914.
Currently we indicate failure/ pass basing on test_results["passed"], and allowed version of preflight/ OCP cluster. 
We should not reimplement this logic in separate task, as in here- you should use some output parameter from step "verify-ci-results".

Reply:
It's not verifying results though, just getting them to set the github status. But I'll take a look at what Harita has on the verify-ci-results task to see how this could be simplified.

Reply:
Updated according to the agreed upon change. Please review again. @haripate just FYI I put some placeholder value in verify-ci-results just so that the pipeline is not broken.